packages/template-retail-react-app/app/pages/product-detail/index.jsx

```diff
@@ -550,6 +615,16 @@ const ProductDetail = () => {
                 selectedStore
             )
 
+            // Fetch and assign a suitable shipment for product items
+            const targetShipmentId = await getShipmentForItems(hasAnyPickupSelected, selectedStore)
```

Comment:
the function only returns a string id, let's make it clear because I am under the impression this func will return a shipment object, not id
```suggestion
            const targetShipmentId = await getShipmentidForItems(hasAnyPickupSelected, selectedStore)
```